src/Aquifer.Common/Utilities/BibleWordIdentifier.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        if (verse is < _minVerse or > _maxVerse)
+        {
+            throw new ArgumentOutOfRangeException(nameof(chapter), chapter, $"{nameof(chapter)} must be between {_minVerse} and {_maxVerse} but was \"{verse}\".");
+        }
+
+        if (word is < _minWord or > _maxWord)
+        {
+            throw new ArgumentOutOfRangeException(nameof(chapter), chapter, $"{nameof(chapter)} must be between {_minWord} and {_maxWord} but was \"{word}\".");
+        }
+
+        if (wordSegment is < _minWordSegment or > _maxWordSegment)
+        {
+            throw new ArgumentOutOfRangeException(nameof(chapter), chapter, $"{nameof(chapter)} must be between {_minWordSegment} and {_maxWordSegment} but was \"{wordSegment}\".");
+        }
+
+        BookId = bookId;
+        Chapter = chapter;
+        Verse = verse;
+        Word = word;
+        WordSegment = wordSegment;
+
+        WordIdentifier = CalculateWordIdentifier((int)bookId - 1, chapter, verse, word, wordSegment);
+    }
+
+    /// <summary>
+    /// Creates a <see cref="BibleWordIdentifier"/> representing the upper bound of a Book.
+    /// </summary>
+    public static BibleWordIdentifier GetUpperBoundOfBook(BookId bookId)
+    {
+        return new BibleWordIdentifier(bookId, _maxChapter, _maxVerse, _maxWord, _maxWordSegment);
+    }
+
+    /// <summary>
+    /// Creates a <see cref="BibleWordIdentifier"/> representing the upper bound of a Chapter.
+    /// </summary>
+    public static BibleWordIdentifier GetUpperBoundOfChapter(BookId bookId, int chapter)
+    {
+        return new BibleWordIdentifier(bookId, chapter, _maxVerse, _maxWord, _maxWordSegment);
+    }
+
+    /// <summary>
+    /// Creates a <see cref="BibleWordIdentifier"/> representing the upper bound of a Verse.
+    /// </summary>
+    public static BibleWordIdentifier GetUpperBoundOfVerse(BookId bookId, int chapter, int verse)
+    {
+        return new BibleWordIdentifier(bookId, chapter, verse, _maxWord, _maxWordSegment);
+    }
+
+    /// <summary>
+    /// Creates a <see cref="BibleWordIdentifier"/> representing the upper bound of a Word.
+    /// </summary>
+    public static BibleWordIdentifier GetUpperBoundOfWord(BookId bookId, int chapter, int verse, int word)
+    {
+        return new BibleWordIdentifier(bookId, chapter, verse, word, _maxWordSegment);
+    }
+
+    public static implicit operator long(BibleWordIdentifier source)
+    {
+        return source.WordIdentifier;
+    }
+
+    public override string ToString()
+    {
+        return WordIdentifier.ToString("D12");
+    }
+
+    private static long CalculateWordIdentifier(int book, int chapter, int verse, int word, int wordSegment)
+    {
+        return (book * _bookMultiplier) + (chapter * _chapterMultiplier) + (verse * _verseMultiplier) + (word * _wordMultiplier) + wordSegment;
+    }
+
+    private static readonly BookId _maxBookId = Enum.GetValues(typeof(BookId)).Cast<BookId>().Max();
```

Comment:
Same message about the consts too.
![image](https://github.com/user-attachments/assets/8cb9dd81-fd50-41fa-a6b2-155ae2111073)

Reply:
Where is this rule defined?  I'm not seeing it in the `.editorconfig` so I'm not sure where Rider is getting that from.

That's a surprising naming convention because public properties are also Pascal case and they would look the same.  I've usually done public `const` values in Pascal and private const values in a different convention to help distinguish between them visually.

Reply:
I updated my PR to follow Rider's suggested styling and will look into updating the `.editorconfig` to enforce it when I get some downtime.